### PR TITLE
Use codecov orb over local script

### PR DESCRIPTION
**Pull request by eessex**
Some cleanup to coverage reporting, this pr demonstrates a model for uploading coverage reports from multiple test runners.

- Use the `codecov` orb instead of saving this package's contents locally in `/scripts/codecov`
- Let codecov handle merging reports on their site and remove transformation logic
- Move `nyc` config to `package.json`
- Use `hokusai/test` step from orb over `yarn test` in ci

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,14 +18,12 @@
     "start": "sh scripts/start.sh",
     "production": "sh scripts/production.sh",
     "publish-assets": "sh scripts/publish-assets.sh",
+    "report-coverage": "./node_modules/.bin/codecov -c $COMMIT_HASH_FULL -b $BRANCH_NAME",
```
Comment (damassi) [inline package.json]: 👌

**Comment by damassi** [comment]
@eessex - Can this be merged?

**Comment by eessex** [comment]
Something is not connecting quite right, will follow up on this soon.

File: Dockerfile [hunk 2]
```diff
@@ -10,7 +10,7 @@ RUN apk add --no-cache --quiet \
   bash \
   curl \
   dumb-init \
-  git && \
+  git mercurial && \
```
Comment (damassi) [inline Dockerfile]: What is mercurial?

Comment (eessex) [reply Dockerfile]: Mercurial provides the lib `hg` which is used by codecov's upload script-- however we did not need it in the end.

**Pull request merged**